src/mono/wasm/runtime/driver.c

```diff
@@ -601,21 +601,25 @@ mono_wasm_assembly_load (const char *name)
 	return res;
 }
 
-EMSCRIPTEN_KEEPALIVE MonoClass* 
-mono_wasm_find_corlib_class (const char *namespace, const char *name)
+EMSCRIPTEN_KEEPALIVE MonoAssembly* 
+mono_wasm_get_corlib ()
 {
-	return mono_class_from_name (mono_get_corlib (), namespace, name);
+	return mono_image_get_assembly (mono_get_corlib());
 }
 
 EMSCRIPTEN_KEEPALIVE MonoClass*
 mono_wasm_assembly_find_class (MonoAssembly *assembly, const char *namespace, const char *name)
 {
+	if (!assembly)
+		return NULL;
```

Comment:
here and elsewhere:

why do you want a soft failure?  It seems like it's always a logic error to pass `NULL` to any of these functions. ie, why not:
```c
  g_assert (assembly);
```